public/app/features/teams/TeamGroupSync.tsx

```diff
@@ -1,93 +1,70 @@
 import { css, cx } from '@emotion/css';
-import { FormEventHandler, PureComponent } from 'react';
-import { connect, ConnectedProps } from 'react-redux';
+import { FormEventHandler, useState } from 'react';
 
+import { TeamGroupDto } from '@grafana/api-clients/rtkq/legacy';
 import { Trans, t } from '@grafana/i18n';
 import { Input, Tooltip, Icon, Button, useTheme2, InlineField, InlineFieldRow } from '@grafana/ui';
 import { SlideDown } from 'app/core/components/Animations/SlideDown';
 import { CloseButton } from 'app/core/components/CloseButton/CloseButton';
 import EmptyListCTA from 'app/core/components/EmptyListCTA/EmptyListCTA';
 import { UpgradeBox, UpgradeContent, UpgradeContentProps } from 'app/core/components/Upgrade/UpgradeBox';
 import { highlightTrial } from 'app/features/admin/utils';
-import { StoreState } from 'app/types/store';
-import { TeamGroup } from 'app/types/teams';
 
-import { addTeamGroup, loadTeamGroups, removeTeamGroup } from './state/actions';
-import { getTeamGroups } from './state/selectors';
+import { useAddTeamGroup, useGetTeamGroups, useRemoveTeamGroup } from './hooks';
 
-function mapStateToProps(state: StoreState) {
-  return {
-    groups: getTeamGroups(state.team),
-  };
-}
-
-const mapDispatchToProps = {
-  loadTeamGroups,
-  addTeamGroup,
-  removeTeamGroup,
-};
-
-interface OwnProps {
+interface Props {
   isReadOnly: boolean;
   teamUid: string;
 }
 
-interface State {
-  isAdding: boolean;
-  newGroupId: string;
-}
-
-const connector = connect(mapStateToProps, mapDispatchToProps);
-export type Props = OwnProps & ConnectedProps<typeof connector>;
-
 const headerTooltip = `Sync LDAP, OAuth or SAML groups with your Grafana teams.`;
 
-export class TeamGroupSync extends PureComponent<Props, State> {
-  constructor(props: Props) {
-    super(props);
-    this.state = { isAdding: false, newGroupId: '' };
-  }
-
-  componentDidMount() {
-    this.fetchTeamGroups();
-  }
+export const TeamGroupSync = ({ isReadOnly, teamUid }: Props) => {
+  const [isAdding, setIsAdding] = useState(false);
```

Comment:
We can probably use the loading states from the two mutations to check if one is being added or removed, rather than tracking the state separately.

Something like

```tsx
const [addTeamGroup, addingGroupResponse] = useAddTeamGroup();
const [removeTeamGroup, removingGroupResponse] = useRemoveTeamGroup();

const isAddingOrRemoving = addingGroupResponse.isLoading || removingGroupResponse.isLoading;
```